tokio/src/io/driver/scheduled_io.rs

```diff
@@ -214,7 +215,10 @@ impl ScheduledIo {
     fn wake0(&self, ready: Ready, shutdown: bool) {
         const NUM_WAKERS: usize = 32;
 
-        let mut wakers: [Option<Waker>; NUM_WAKERS] = Default::default();
+        let mut wakers: [MaybeUninit<Option<Waker>>; NUM_WAKERS] = unsafe {
+          core::mem::MaybeUninit::uninit().assume_init()
+        };
```

Comment:
If a panic happens anywhere in this code, then the wakers in this array are leaked. Can you make a new file in `src/util` that combines the array and counter into a struct with a safe API and a proper destructor that actually drops them?

Additionally, I happen to know that we use this strategy several places in Tokio.